### DISCUSSION TITLE
Disconnecting explicit many-to-many

### DISCUSSION THREAD

**Discussion by ys947** [prisma/prisma#13620] [Q&A]
Hi,
From my understanding to connect existing records in an explicit many-to-many relationship the code would be something like
```
await prisma.post.update({
  where: {
    id: post.id,
  },
  data: {
    tags: {
      create: [
        { tag: { connect: { name: 'dev' } } },
        { tag: { connect: { name: 'prisma' } } },
      ],
    },
  },
})
```
I'm a bit confused on what is the best practice for disconnecting existing records in an explicit many-to-many relationship?

**Comment by austincrim** [comment]
Hey @ys947 👋,

You can use the [`disconnect`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#disconnect) option to accomplish this:
```ts
await prisma.post.update({
  where: { id: 1 },
  data: {
    tags: {
      disconnect: [{ id: 1 }, { id: 2 }]
    }
  }
})
```

**Reply by ys947** [reply]
Hey,
Thank you for your response, unless I'm misunderstanding something this solution seems to work for implicit many to many relationships but not explicit ones?

**Reply by austincrim** [reply]
Oh, sorry, I missed that part of your question. The process should be the same, though!

```ts
await prisma.post.update({
  where: { id: 1 },
  data: {
    tagsOnPosts: {
      disconnect: [
        {
          tagId_postId: {
            postId: 1,
            tagId: 1
          }
        },
        {
          tagId_postId: {
            postId: 1,
            tagId: 2
          }
        }
      ]
    }
  }
})
```

Let me know if you run into problems with that approach.

**Reply by ys947** [reply]
I have tried it and I am getting `The change you are trying to make would violate the required relation...`, although that could be relating to my specific implementation but if it is very similar to the example.

I did manage to get it working like this, I'm not sure if that is best approach

```
await prisma.TagsOnPosts.deleteMany({
  where: {
    OR: [
      { postId: post.id, tagId: 1 },
      { postId: post.id, tagId: 2 },
    ],
  },
})
```

I can create a new project with just the example data and test it again to confirm if the error is due to something specific to my project?

**Reply by austincrim** [reply]
Could you share your application code and maybe some sample data so I can try to recreate your error?

**Reply by ys947** [reply]
I have created the example tables now
```
datasource db {
    provider = "postgresql"
    url      = env("DATABASE_URL")
}

generator client {
    provider = "prisma-client-js"
}

model Post {
    id    Int           @id @default(autoincrement())
    title String
    tags  TagsOnPosts[]
}

model Tag {
    id    Int           @id @default(autoincrement())
    name  String
    posts TagsOnPosts[]
}

model TagsOnPosts {
    post       Post     @relation(fields: [postId], references: [id])
    postId     Int
    tag        Tag      @relation(fields: [tagId], references: [id])
    tagId      Int
    assignedAt DateTime @default(now())
    assignedBy String

    @@id([postId, tagId])
}
```

Using the code you suggested (with replacing `data: { tagsOnPosts: {` with `data: { tags: {`) I get the error `The change you are trying to make would violate the required relation 'PostToTagsOnPosts' between the 'Post' and 'TagsOnPosts' models.`

**Reply by austincrim** [reply]
Your error message jogged my memory that `disconnect` is [only available on **optional** relations](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#remarks-17). Sorry for the confusion on my part! 

Along with that, list relations cannot be optional, so I don't think `disconnect` is an option here.

If you are fine with deleting out of the explicit join table, go ahead with that workaround 😄.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
